docs/defaults-detailed.rst

```diff
@@ -9,169 +11,640 @@ them.
+  parameter to ``False``, the IPv4 configuration will be disabled.
+
+``inet6``
+  Optional. IPv6 configuration method used by a given interface. There are many
+  configuration methods described in the :manpage:`interfaces(5)` manual page, most
+  commonly used are: ``auto``, ``manual``, ``dhcp``, ``static``, ``v4tunnel``,
+  ``6to4``. If you set this parameter to ``False``, the IPv4 configuration will
+  be disabled.
+
+``address`` or ``addresses``
+  Optional. A string or an YAML list of IPv4 and/or IPv6 addresses to set on
+  a given network interface, in the form of ``ipaddress/prefix`` or CIDR.
+  Remember that you need to specify the host IP address and not the network;
+  the ``192.0.2.1/24`` is the correct notation, and ``192.0.2.0/24`` is
+  incorrect.
+
+``gateway`` or ``gateway4``
+  Optional. Specify the IPv4 address of the network gateway to which outgoing
+  packets will be directed.
+
+``gateway6``
+  Optional. Specify the IPv6 address of the network gateway to which outgoing
+  packets will be directed.
+
+DNS nameserver and search parameters
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+``dns_nameservers``
+  Optional. String or list of IP addresses of the nameservers to configure in
+  :file:`/etc/resolv.conf`. Remember that only 3 nameservers are allowed at any
+  time. They will be added to the IPv4 section of the network interface
+  configuration unless IPv4 is disabled, in which case they will be configured
+  in IPv6 section.
+
+``dns_search``
+  Optional. String or list of domains which should be searched in the DNS if
+  a hostname without a domain is specified. They will be added to the
+  :file:`/etc/resolv.conf`. This list will be added to the IPv4 section of the
+  network interface configuratio unless IPv4 is disabled, in which case they
+  will be configured in IPv6 section.
+
+``dns_nameservers4``
+  Optional. String or list of IPv4 addresses of the nameservers to configure in
+  :file:`/etc/resolv.conf`. Remember that only 3 nameservers are allowed at any
+  time. They will be added to the IPv4 section of the network interface
+  configuration.
+
+``dns_search4``
+  Optional. String or list of domains which should be searched in the DNS if
+  a hostname without a domain is specified. They will be added to the
+  :file:`/etc/resolv.conf`. This list will be added to the IPv4 section of the
+  network interface configuratio.
+
+``dns_nameservers6``
+  Optional. String or list of IPv6 addresses of the nameservers to configure in
+  :file:`/etc/resolv.conf`. Remember that only 3 nameservers are allowed at any
+  time. They will be added to the IPv6 section of the network interface
+  configuration.
+
+``dns_search6``
+  Optional. String or list of domains which should be searched in the DNS if
+  a hostname without a domain is specified. They will be added to the
+  :file:`/etc/resolv.conf`. This list will be added to the IPv6 section of the
+  network interface configuratio.
```

Comment:
DNS nameserver and search domains aren't really tied to IPv4/IPv6, so I guess we could make the role simplier and drop the `4`/`6` suffix from the variables. Agreed.